### PR TITLE
Ops 441 update user details

### PR DESCRIPTION
**Pull request by johndeange**
## What changed

- Implement `USER_ADMIN` role changing `User` details.
- Deactivate all `UserSession` when changing `UserStatus` to `INACTIVE/LOCKED`

## Issue

https://github.com/HHS/OPRE-OPS/issues/441

## How to test

1. Login as Budget Team Member
2. Navigate to the `User Admin` page.
3. Search for and select User Demo user
4. Update division, roles and set User Demo status to `LOCKED`
5. Log out and Log in with User Demo (Basic) user.
6. Log in should fail and an event should be created in the DB for a failed login.
7. Log in as Budget Team Member
8. Set User Demo status to `ACTIVE`
9. Log out and you should be able to log in with User Demo (Basic) now.

## Screenshots

image

## Definition of Done Checklist
- [x] OESA: Code refactored for clarity
- [x] OESA: Dependency rules followed
- [x] Automated unit tests updated and passed
- [x] Automated integration tests updated and passed
- [x] Automated quality tests updated and passed
- [x] Automated load tests updated and passed
- [x] Automated a11y tests updated and passed
- [x] Automated security tests updated and passed
- [x] 90%+ Code coverage achieved
- [x] Form validations updated

### CODE REVIEW

File: frontend/src/components/Users/UserEmailComboBox/UserEmailComboBox.jsx [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
-function UserEmailComboBox({ legendClassName = "usa-label margin-top-0" }) {
-    const [selectedUsers, setSelectedUsers] = React.useState([]);
+/**
+ * Renders a ComboBox for selecting users by email.
+ * @param {Array<any>} selectedUsers - The selected users.
+ * @param {Function} setSelectedUsers - The function to set the selected users.
+ * @returns {JSX.Element} - The rendered component.
+ */
+function UserEmailComboBox({ selectedUsers, setSelectedUsers }) {
+    // const [selectedUsers, setSelectedUsers] = React.useState([]);
```
Comment (fpigeonjr) [inline frontend/src/components/Users/UserEmailComboBox/UserEmailComboBox.jsx:12]: pls remove commented code

File: frontend/src/components/Users/UserInfo/UserInfo.jsx [hunk 2]
```diff
@@ -16,6 +23,7 @@ const UserInfo = ({ user, isEditable }) => {
 
     const { data: divisions, error: errorDivisions, isLoading: isLoadingDivisions } = useGetDivisionsQuery();
     const { data: roles, error: errorRoles, isLoading: isLoadingRoles } = useGetRolesQuery();
+    const [updateUser, updateUserResult] = useUpdateUserMutation();
```
Comment (fpigeonjr) [inline frontend/src/components/Users/UserInfo/UserInfo.jsx:26]: ⭐ thx for sharing this pattern for handling result from the api call

File: frontend/src/components/Users/UserInfo/UserInfo.test.js [hunk 3]
```diff
@@ -162,6 +162,108 @@ describe("UserInfo", () => {
...
+        });
+
+        expect(await screen.findByText("User Details")).toBeInTheDocument(); // Card Header
+
+        // find the input element within the div with testid division-combobox
+        const divisionComboBox = screen.getByTestId("division-combobox");
+        // eslint-disable-next-line testing-library/no-node-access
+        const divisionInput = divisionComboBox.querySelector("input");
+
+        fireEvent.keyDown(divisionInput, { key: "ArrowDown", code: 40 });
```
Comment (fpigeonjr) [inline frontend/src/components/Users/UserInfo/UserInfo.test.js:192]: learned that the `[userEvent](https://testing-library.com/docs/user-event/intro/)` is now the preferred method of handling user interactions

File: frontend/src/pages/users/admin/UserAdmin.jsx [hunk 4]
```diff
@@ -1,26 +1,37 @@
 import App from "../../../App";
 import UserEmailComboBox from "../../../components/Users/UserEmailComboBox/index.js";
-import { useGetDivisionsQuery } from "../../../api/opsAPI.js";
+import React from "react";
+import UserInfo from "../../../components/Users/UserInfo/UserInfo.jsx";
```
Comment (fpigeonjr) [inline frontend/src/pages/users/admin/UserAdmin.jsx:4]: don't need the double UserInfo on the import

`../../../components/Users/UserInfo"`

**Review by fpigeonjr — APPROVED** [review]
🥇 looks good @johndeange, made some non-blocking suggestions. Would be nice to add some UX like confirmation Alert (if changes were applied)and Cancel/Save buttons to allow user to back out of changes.

File: frontend/src/components/Users/UserInfo/UserInfo.jsx [hunk 5]
```diff
@@ -29,14 +37,31 @@ const UserInfo = ({ user, isEditable }) => {
...
     if (isLoadingDivisions || isLoadingRoles) {
         return <div>Loading...</div>;
     }
     if (errorDivisions || errorRoles) {
         return <div>Oops, an error occurred</div>;
     }
-
-    console.log("isEditable: ", isEditable);
+    if (updateUserResult.isError) {
+        return <div>Oops, an error occurred</div>;
```
[the remark this answers is not in the document]
Comment (johndeange) [reply frontend/src/components/Users/UserInfo/UserInfo.jsx:63]: It is - the div is displayed in the case of an API error

File: frontend/src/api/opsAPI.js [hunk 6]
```diff
@@ -184,13 +185,13 @@ export const opsApi = createApi({
             invalidatesTags: ["User"]
         }),
         updateUser: builder.mutation({
-            query: ({ id, body }) => ({
+            query: ({ id, data }) => ({
                 url: `/users/${id}`,
-                method: "PUT",
+                method: "PATCH",
```
Comment (fpigeonjr) [inline frontend/src/api/opsAPI.js:190]: 🤔 does the openAPI spec need to be updated to match?

File: frontend/src/components/Users/UserInfo/UserInfo.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (fpigeonjr) [reply frontend/src/components/Users/UserInfo/UserInfo.jsx:63]: ok I guess I was envisioning an error Alert with a message to try again or the div to be displayed above the User card.

**Review by maiyerlee — APPROVED** [review]
I followed the 'How to Test' steps, and the changes worked as expected.

**Pull request merged**